echo_request_handler.cc

```diff
@@ -0,0 +1,41 @@
+#include "echo_request_handler.h"
+#include <string>
+#include <iostream>
+
+const int MAX_REQ_SIZE = 8192;
+
+EchoRequestHandler::EchoRequestHandler(boost::shared_ptr<bai::tcp::socket> socket)
+  : socket_(socket)
+{
+}
+
+EchoRequestHandler::~EchoRequestHandler() {
+}
+
+// Read in the request from the socket used to construct the handler, then
+// fill the response buffer with the necessary headers followed by the
+// original request.
+char* EchoRequestHandler::constructEcho() {
```

Comment:
It doesn't look like `EchoRequestHandler` requires a full-fledged class? We only need need to construct a response buffer that echoes back the received request, which is a single method. We could inline this definition in a current file, but we will eventually have several different request handlers. Thus, it's worth grouping them. 

Have you considered creating a new `request_handlers.h` header which defines the following? 

```c++
namespace lightning_server {
     namespace request_handlers {
         char* echoRequestHandler(boost::shared_ptr<bai::tcp::socket> socket)
     }
}
```

This could then be implemented in a `request_handlers.cc` file: 

```c++
using lightning_server::request_handlers;

char* request_handlers::echoRequestHandler(boost::shared_ptr<bai::tcp::socket> socket) {
     // ...
}
```

And so on for other request handlers.